### PR TITLE
API test for getWebDAVSharePermissions

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds more tests for get webDAV share permissions
## Related Issue
#31531 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiSharing-v1/getWebDAVSharePermissions.feature [hunk 1]
```diff
@@ -38,7 +68,22 @@ Feature: sharing
 		And user "user0" has shared file "tmp.txt" with user "user1"
 		When user "user0" updates the last share using the API with
 			| permissions | 17 |
-		And user "user1" gets the following properties of folder "/tmp.txt" using the API
+		And user "user1" gets the following properties of file "/tmp.txt" using the API
+			|{http://open-collaboration-services.org/ns}share-permissions |
+		Then the single response should contain a property "{http://open-collaboration-services.org/ns}share-permissions" with value "17"
+
+	Scenario: Correct webdav share-permissions for received file with reshare permissions but no edit permissions
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/getWebDAVSharePermissions.feature]: "received group shared file"

File: tests/acceptance/features/apiSharing-v1/getWebDAVSharePermissions.feature [hunk 2]
```diff
@@ -16,7 +16,22 @@ Feature: sharing
 		And user "user1" has been created
 		And user "user0" has uploaded file with content "foo" to "/tmp.txt"
 		And user "user0" has shared file "/tmp.txt" with user "user1"
-		When user "user1" gets the following properties of folder "/tmp.txt" using the API
+		When user "user1" gets the following properties of file "/tmp.txt" using the API
+			|{http://open-collaboration-services.org/ns}share-permissions |
+		Then the single response should contain a property "{http://open-collaboration-services.org/ns}share-permissions" with value "19"
+
+	Scenario: Correct webdav share-permissions for received group share file with edit and reshare permissions
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing-v1/getWebDAVSharePermissions.feature]: s/share file/shared file/g

**Review by phil-davis — COMMENTED** [review]
just words in scenario descriptions

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on #31575

**Comment by paurakhsharma** [comment]
Backport on #31575

**Pull request merged**